vortex-file/src/read/mod.rs

```diff
@@ -57,28 +59,19 @@ pub type MessageId = Vec<LayoutPartId>;
 /// the message contents.
 #[derive(Debug, Clone)]
 pub struct MessageLocator(pub MessageId, pub ByteRange);
+/// A message that has had its bytes materialized onto the heap.
+#[derive(Debug, Clone)]
+pub struct Message(pub MessageId, pub Bytes);
 
-#[derive(Debug)]
-pub enum BatchRead {
-    ReadMore(Vec<MessageLocator>),
-    Batch(ArrayData),
-}
-
-#[derive(Debug)]
-pub enum MetadataRead {
-    /// Layout has no metadata
-    None,
-    /// Additional IO is required
-    ReadMore(Vec<MessageLocator>),
-    Batches(Vec<Option<ArrayData>>),
-}
+pub type BatchRead = MessageRead<ArrayData>;
```

Comment:
I personally find these type aliases more confusing than using the underlying type, though I think that is controversial